### PR TITLE
GNOME - Better Opening Of Files On Startup

### PR DESCRIPTION
**Pull request by nlogozzo**

### CODE REVIEW

File: NickvisionMoney.GNOME/Views/MainWindow.cs [hunk 1]
```diff
@@ -384,6 +373,20 @@ private async void AccountAdded(object? sender, EventArgs e)
...
+    /// <summary>
+    /// Occurs when the window is shown
+    /// </summary>
+    /// <param name="sender">Gtk.Widget</param>
+    /// <param name="e">EventArgs</param>
+    private async void OnShow(Gtk.Widget sender, EventArgs e)
+    {
+        if (_controller.FileToLaunch != null)
+        {
+            await _controller.AddAccountAsync(_controller.FileToLaunch);
```
Comment (fsobolev) [inline NickvisionMoney.GNOME/Views/MainWindow.cs:385]: Just a note: I guess when GLib.AddTimeout will be implemented we'll be able to fix animation by wrapping this line in that function with ~500ms delay added.

**Review by fsobolev — APPROVED** [review]

File: NickvisionMoney.GNOME/Views/MainWindow.cs (hunk 1 above)
Comment (nlogozzo) [reply NickvisionMoney.GNOME/Views/MainWindow.cs:385]: Just thought of this...can you marshall `g_timeout_add()`?

Comment (fsobolev) [reply NickvisionMoney.GNOME/Views/MainWindow.cs:385]: I'm not sure how to properly write a delegate for this...

Comment (nlogozzo) [reply NickvisionMoney.GNOME/Views/MainWindow.cs:385]:
```csharp
private delegate bool GSourceFunc(nint data);

[LibraryImport("libadwaita-1.so.0", StringMarshalling = StringMarshalling.Utf8)]
private static partial uint g_timeout_add(uint interval, [MarshalAs(UnmanagedType.FunctionPtr)]GSourceFunc function, nint data);
```

File: NickvisionMoney.GNOME/Program.cs [hunk 2]
```diff
@@ -95,7 +95,7 @@ public int Run(string[] args)
     /// </summary>
     /// <param name="sedner">Gio.Application</param>
     /// <param name="e">EventArgs</param>
-    private async void OnActivate(Gio.Application sedner, EventArgs e)
+    private void OnActivate(Gio.Application sedner, EventArgs e)
```
Comment (DaPigGuy) [inline NickvisionMoney.GNOME/Program.cs:98]:
```suggestion
    /// <param name="sender">Gio.Application</param>
    /// <param name="e">EventArgs</param>
    private void OnActivate(Gio.Application sender, EventArgs e)
 ```

Comment (DaPigGuy) [reply NickvisionMoney.GNOME/Program.cs:98]: Oh I commented after merge :sweat_smile:

Comment (fsobolev) [reply NickvisionMoney.GNOME/Program.cs:98]: thanks anyway, will be fixed in some other PR later :)

Comment (nlogozzo) [reply NickvisionMoney.GNOME/Program.cs:98]: Haha no worries :) fixed https://github.com/nlogozzo/NickvisionMoney/commit/7eeaffa7ff26d39f2cdaef3b24bf35565826bdd5

**Pull request merged**